### PR TITLE
Gui/gui trantorians classes implementation

### PR DESCRIPTION
**Pull request by Klayni**
- Add Trantorian class
- Add Tribe class
- Add Action enum and Action class
- Add unit tests

### CODE REVIEW

**Review by 6im0n — APPROVED** [review]
Build not work, when it's fix it will be good for me :+1:

File: gui/src/position/Position.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+using namespace GUI;
+
+Position::Position(unsigned int x, unsigned int y)
+    :   _x(x)
+    ,   _y(y)
+{
+}
+
+Position::~Position()
```
Comment (Quentintnrl) [inline gui/src/position/Position.cpp:16]: Why it's not set to default ?

File: gui/src/trantorians/Actions.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+** Actions
+*/
+
+#include "Actions.hpp"
+
+GUI::Actions::Actions()
+{
+}
+
+GUI::Actions::~Actions()
```
Comment (Quentintnrl) [inline gui/src/trantorians/Actions.cpp:12]: Same here, what it's not set to default ?

File: gui/src/trantorians/Trantorian.cpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    _team = teamName;
+    _position.setX(x);
+    _position.setY(y);
+    _alive = true;
+    _action = false;
+    _level = 1;
+    _lifetimeRemaining = 100;
+}
+
+GUI::Trantorian::~Trantorian()
```
Comment (Quentintnrl) [inline gui/src/trantorians/Trantorian.cpp:19]: Same here ?

File: gui/src/trantorians/Tribes.cpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+** Tribes
+*/
+
+#include "Tribes.hpp"
+
+GUI::Tribes::Tribes()
+{
+}
+
+GUI::Tribes::~Tribes()
```
Comment (Quentintnrl) [inline gui/src/trantorians/Tribes.cpp:12]: Same here ?

File: tests/gui/trantorian_methods.cpp
Comment (Quentintnrl) [inline tests/gui/trantorian_methods.cpp:1]: Why this file isn't name like the other file ?

File: tests/gui/tribes_methods.cpp
Comment (Quentintnrl) [inline tests/gui/tribes_methods.cpp:1]: same here ?

File: gui/src/main.cpp
Comment (BxptisteM) [inline gui/src/main.cpp:1]: Don't remove this :
```
** EPITECH PROJECT, 2024
** Zappy
```

File: gui/src/trantorians/Actions.cpp
Comment (BxptisteM) [inline gui/src/trantorians/Actions.cpp:1]: Fix the header please

File: gui/src/trantorians/Actions.hpp
Comment (BxptisteM) [inline gui/src/trantorians/Actions.hpp:1]: Fix the header

File: gui/src/trantorians/Actions.hpp [hunk 5]
```diff
@@ -0,0 +1,103 @@
+/*
+** File description:
+** Actions
+*/
+
+#ifndef ACTIONS_HPP_
+#define ACTIONS_HPP_
```
Comment (BxptisteM) [inline gui/src/trantorians/Actions.hpp:7]: Indent this

File: gui/src/trantorians/Trantorian.cpp
Comment (BxptisteM) [inline gui/src/trantorians/Trantorian.cpp:1]: fix the header

File: gui/src/trantorians/Trantorian.hpp
Comment (BxptisteM) [inline gui/src/trantorians/Trantorian.hpp:1]: Same as the Action.hpp file

**Review by BxptisteM — APPROVED** [review]

**Review by RenardFute — APPROVED** [review]

**Pull request merged**